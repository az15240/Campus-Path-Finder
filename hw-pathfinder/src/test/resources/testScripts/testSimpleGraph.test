# Tests the Dijkstra algorithm on a simple graph with five node

CreateGraph g

AddNode g A
AddNode g B
AddNode g C
AddNode g D
AddNode g E

AddEdge g A B 1
AddEdge g A C 2
AddEdge g A D 2
AddEdge g B D 1
AddEdge g C D 3
AddEdge g B C 10
AddEdge g D E 2

ListNodes g

ListChildren g A

FindPath g A B
FindPath g A C
FindPath g A D
FindPath g A E
