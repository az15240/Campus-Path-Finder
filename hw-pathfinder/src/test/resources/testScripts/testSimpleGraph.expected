# Tests the Dijkstra algorithm on a simple graph with five node

created graph g

added node A to g
added node B to g
added node C to g
added node D to g
added node E to g

added edge 1.000 from A to B in g
added edge 2.000 from A to C in g
added edge 2.000 from A to D in g
added edge 1.000 from B to D in g
added edge 3.000 from C to D in g
added edge 10.000 from B to C in g
added edge 2.000 from D to E in g

g contains: A B C D E

the children of A in g are: B(1.000) C(2.000) D(2.000)

path from A to B:
A to B with weight 1.000
total cost: 1.000
path from A to C:
A to C with weight 2.000
total cost: 2.000
path from A to D:
A to D with weight 2.000
total cost: 2.000
path from A to E:
A to D with weight 2.000
D to E with weight 2.000
total cost: 4.000